api/python/quilt3/admin/__init__.py

```diff
@@ -0,0 +1,178 @@
+role_adapter = pydantic.TypeAdapter(AnnotatedRole)
+
+
+@pydantic.dataclasses.dataclass
+class User:
+    name: str
+    email: str
+    date_joined: datetime
+    last_login: datetime
+    is_active: bool
+    is_admin: bool
+    is_sso_only: bool
+    is_service: bool
+    role: Optional[AnnotatedRole]
+    extra_roles: List[AnnotatedRole]
+
+
+class Quilt3AdminError(Exception):
+    def __init__(self, details):
+        super().__init__(details)
+        self.details = details
+
+
+class UserNotFoundError(Quilt3AdminError):
+    def __init__(self):
+        super().__init__(None)
+
+
+def _handle_errors(result: _graphql_client.BaseModel) -> Any:
+    if isinstance(result, (_graphql_client.InvalidInputSelection, _graphql_client.OperationErrorSelection)):
+        raise Quilt3AdminError(result)
+    return result
+
+
+def _get_client():
+    return _graphql_client.Client()
+
+
+def get_user(name: str) -> Optional[User]:
+    """
+    Get a specific user from the registry. Return `None` if the user does not exist.
+
+    Args:
+        name: Username of user to get.
+    """
+    result = _get_client().get_user(name=name)
+    if result is None:
+        return None
+    return User(**result.model_dump())
+
+
+def get_users() -> List[User]:
+    """
+    Get a list of all users in the registry.
+    """
+    return [User(**u.model_dump()) for u in _get_client().get_users()]
+
+
+def create_user(name: str, email: str, role: str, extra_roles: Optional[List[str]] = None) -> None:
+    """
+    Create a new user in the registry.
+
+    Args:
+        name: Username of user to create.
+        email: Email of user to create.
+        role: Active role of the user.
+        extra_roles: Additional roles to assign to the user.
+    """
+
+    _handle_errors(
+        _get_client().create_user(input=_graphql_client.UserInput(name=name, email=email, role=role, extra_roles=extra_roles))
+    )
+
+
+def delete_user(name: str) -> None:
+    """
+    Delete user from the registry.
+
+    Args:
+        name: Username of user to delete.
+    """
+    result = _get_client().delete_user(name=name)
+    if result is None:
+        raise UserNotFoundError
+    _handle_errors(result.delete)
+
+
+def get_roles() -> List[Union[_graphql_client.GetRolesRolesUnmanagedRole, _graphql_client.GetRolesRolesManagedRole]]:
+    """
+    Get a list of all roles in the registry.
+    """
+    return [role_adapter.validate_python(r.model_dump()) for r in _get_client().get_roles()]
+
+
+def set_role(
+    name: str,
+    role: str,
+    extra_roles: Optional[List[str]] = None,
+    *,
+    append: bool = False,
+) -> None:
+    """
+    Set the active and extra roles for a user.
+
+    Args:
+        name: Username of user to update.
+        role: Role to be set as the active role.
+        extra_roles: Additional roles to assign to the user.
+        append: If True, append the extra roles to the existing roles. If False, replace the existing roles.
```

Comment:
if you add the comments directly to the arguments, will the doc gen use them?

Reply:
there is no way to put docstrings directly to parameters
that way of documenting parameters is just one of the conventions on top of function docstrings (specifically from [Google one](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings))